src/BloomTests/UpdateVersionTableTests.cs

```diff
@@ -25,6 +25,43 @@ public void ThisVersionTooLarge_ReturnsEmptyString()
 		}
 
 		[Test]
+		public void LookupURLOfUpdate_AllWell_ReportsNoConnectivityError()
+		{
+			var t = new UpdateVersionTable();
+			t.TextContentsOfTable = @"0.0.0,3.1.99999, http://first.com/first";
+			t.RunningVersion = Version.Parse("3.2.0");
+			var lookupResult = t.LookupURLOfUpdate();
+			Assert.IsFalse(lookupResult.IsConnectivityError);
+			Assert.IsNullOrEmpty(lookupResult.URL);
+		}
+
+		[Test]
+		public void UpgradeTableParsingError_NoCrash()
+		{
+			var t = new UpdateVersionTable();
+			t.TextContentsOfTable = @"0.0.0,3,1,99999, http://first.com/first"; // too many commas
+			t.RunningVersion = Version.Parse("3.2.0");
+			var lookupResult = t.LookupURLOfUpdate();
+			Assert.IsNullOrEmpty(lookupResult.URL);
+
+			t = new UpdateVersionTable();
+			t.TextContentsOfTable = @"0.0.0,, http://first.com/first"; // too few commas
+			t.RunningVersion = Version.Parse("3.2.0");
+			lookupResult = t.LookupURLOfUpdate();
+			Assert.IsNullOrEmpty(lookupResult.URL);
```

Comment:
This looks like two independent tests (lookupURLOfUpdate_TooManyCommas_GivesErrorAndNoURL, etc)